### PR TITLE
feat(openapi): Add `updateSingleSpec` option to automatically update an only available spec file without any prompts

### PR DESCRIPTION
**Pull request by shaiarmis**
Today when using the `rdme openapi <path> --version=<version>` command to update a spec file (selecting `Update` in the first prompt), we're prompted for selecting an API spec file, even if there's only a single one present.

## 🧰 Changes

This PR adds the `--updateSingleSpec` option that will instruct the CLI to not prompt the user if there's only a single spec file available, and choose it automatically.

## 🧬 QA & Testing

1. Start with a clean project with no API spec files at all
2. Run `rdme openapi <path> --version=<version>`, which will upload the file without any create/update prompt because there's nothing to update.
3. Run `rdme openapi <path> --version=<version>` once again, and select `update` in the first prompt and expect to see another prompt with a single option.
4. Abort the previous command.
5. Run `rdme openapi <path> --version=<version> --updateSingleSpec` and expect the file to upload successfully without any prompts.

### CODE REVIEW

**Comment by domharrington** [comment]
Thanks for this! I'm gunna assign to @kanadgupta as he's the person primarily responsible for managing this repo. I _think_ we already have some things in the works to help with this, but i'll see what Kanad says.

File: README.md
Comment (kanadgupta) [inline README.md:120]: ooops... great catch 😅

**Review by kanadgupta — CHANGES_REQUESTED** [review]
Thanks for the PR! Can you add two pieces of logic (and corresponding test cases):

- [ ] An error message if a user passes this `--updateSingleSpec` flag but `apiSettingsBody.length` is greater than 1
- [ ] A warning output if a user passes in the `--id` parameter AND the `--updateSingleSpec` flag ([similar to this](https://github.com/readmeio/rdme/blob/f0da91b00de4bb79ef52dbb62b164e1a4d941271/src/cmds/openapi/index.ts#L101-L103))

**Comment by kanadgupta** [comment]
Also—please ignore the failing [GitHub Action Dry Run](https://github.com/readmeio/rdme/runs/7974845596?check_suite_focus=true) step since that's a known issue with external contributors—but if you could fix [the failing snapshots in the Test Suite](https://github.com/readmeio/rdme/runs/7974845798?check_suite_focus=true#step:6:372), that'd be great. Thank you!

**Comment by shaiarmis** [comment]
> Thanks for the PR! Can you add two pieces of logic (and corresponding test cases):
> 
> * [ ]  An error message if a user passes this `--updateSingleSpec` flag but `apiSettingsBody.length` is greater than 1
> * [ ]  A warning output if a user passes in the `--id` parameter AND the `--updateSingleSpec` flag ([similar to this](https://github.com/readmeio/rdme/blob/f0da91b00de4bb79ef52dbb62b164e1a4d941271/src/cmds/openapi/index.ts#L101-L103))

Done x 2

File: src/cmds/openapi/index.ts
Comment (erunion) [inline src/cmds/openapi/index.ts:107]:
```suggestion
        'When using the `--id` option the `--updateSingleSpec` option is ignored as the desired spec is already specified.'
```

**Review by erunion — COMMENTED** [review]
thanks for this!

**Review by kanadgupta — APPROVED** [review]
Made a few copy edits and added a tiny error case. Thanks again for the PR @shaiarmis!

**Comment by shaiarmis** [comment]
@kanadgupta thanks for the additional changes, they make a lot of sense :)
Should I wait for @erunion's approval before merging this PR?

**Comment by shaiarmis** [comment]
LOL just realized I don't have permissions to merge ^_^ So go ahead and merge at your leisure.

**Review by erunion — APPROVED** [review]

**Pull request merged**
